internal/common/shared/common_pool.go

```diff
@@ -0,0 +1,15 @@
+package shared
+
+import "math/rand"
+
+type baseCommonPool struct {
+	amount int
+}
+
+func (cp *baseCommonPool) checkEnoughInCommonPool() int {
+	return rand.Intn(1)
```

Comment:
Same thing as my comment above about "placeholder" implementations.

Reply:
deleting this file - unused!